R/run_app.R

```diff
@@ -2,15 +2,28 @@
 #'
 #' @inheritParams shared-params
 #' @export
-run_app <- function(snapshot = NULL) {
-  if (is.null(snapshot)) {
-    snapshot <- gsmApp::read_snapshot()
-  }
+
+run_app <- function(
+    dfResults = NULL,
+    dfGroups = NULL,
+    dfMetrics = NULL,
+    dfBounds = NULL) {
+  dfResults <- dfResults %||% gsm::reportingResults %>%
+    dplyr::filter(.data$SnapshotDate == max(.data$SnapshotDate))
+  dfGroups <- dfGroups %||% gsm::reportingGroups
+  dfMetrics <- dfMetrics %||% gsm::reportingMetrics
+  dfBounds <- dfBounds %||% gsm::reportingBounds %>%
+    dplyr::filter(.data$SnapshotDate == max(.data$SnapshotDate))
```

Comment:
I don't think we should give these default values. I made some readability edits, but, as I get deeper into this, these should either have default values, should be required, or should really be `NULL`able (where the `NULL` really means "I am not supplying a value and none should be used`). See [Hadley's in-progress Tidy Design Principles book](https://design.tidyverse.org/inputs-explicit.html) for some tips on how to make functions easier to follow via careful default args.

More specifically, while we'll use `gsm::reportingResults` in examples, the app should fail to launch if the user doesn't supply that argument, so there shouldn't be a default value. `dfBounds` is probably `NULL`-able (because it looks like we ignore it downstream if it isn't supplied), but the app should fail if any of the others are missing.

Reply:
I didn't split this off into a separate issue, since we're defining the usage here. I don't think we should do this "use the sample data if they don't supply data" idea, even if we don't get these "perfect" just yet.

Reply:
Let me re-phrase this to make sure I understand:

- We don't think we should have `dfResults = NULL` in the function deftauls
- We don't think we should load the sample data via the run_app function
- We should only include the sample data in the example of the function documentation

If that's the case, would you be fine with having this as a ticket for later?

Those sample data sets in `run_app` make it easier for me to test changes iteratively. I think it would be good to have a reminder to remove it later as we are closer towards completion.

Reply:
Maybe add a development parameter that defaults to `FALSE` and pulls in testing data otherwise.

Reply:
This will work for now! As I sort out the way we ingest data, we might have a version that accepts reactives, which makes things even more complicated... so let's not try to shoehorn this fix in here!